### ISSUE TITLE
Expand Bootstrap Channel to allow Custom CNI Manifests

### ISSUE THREAD

**Issue by daviddyball** [kubernetes/kops#8142]
In the bootstrap channel builder code ([here](https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/bootstrapchannelbuilder.go#L639)) there is a comment with regard to creating a config option to allow `networking: {"cni": {}}` in cluster spec to use a custom manifest to deploy the CNI. I'd love to see this fleshed out some more, so people could have a more bespoke CNI configuration if necessary without having to bring up a cluster in a broken state (the first time) and then apply their custom CNI.

The [`CNINetworkingSpec`](https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/networking.go#L54) struct could be expanded to include a reference to a manifest stored in `<state-store>/addons/.....` that would then get baked into the `bootstrap-channel.yaml`, much like currently happens for other Networking specs. Granted, it wouldn't allow much in the way of templating, but at least it would allow users to pre-upload their manifest to the state store and reference it before running `kops update cluster` and regenerating `bootstrap-channel.yaml`.

Perhaps something like this?

```
spec:
  networking:
    cni:
      manifest: my.custom-cni.io/manifest.yaml
```

Which would then build `<state-store>/addons/my.custom-cni.io/manifest.yaml` into the `bootstrap-channel.yaml` file and allow a custom CNI to be automatically deployed.

**Comment by olemarkus** [comment]
This is not a bad idea. It is a bit too much for me to take on, but would be happy to review a PR on this.
/kind feature

**Comment by daviddyball** [comment]
This would still be a very useful feature to have.

**Comment by johngmyers** [comment]
This is getting some traction in #14834 
/remove-lifecycle stale

**Comment by daviddyball** [comment]
#14834 was closed in favour of #14955
